### PR TITLE
Crisisline component

### PR DESCRIPTION
**Pull request by SebiLimbu**

### CODE REVIEW

File: src/components/CrisisLineItem.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+import iss from '../iss';
+
+var palette = colors.getPalette();
+
+/*::`*/@reactMixin.decorate(Router.Navigation)/*::`;*/
+class CrisisLineItem extends React.Component {
+    constructor(props: Object) {
+        super(props);
+        this.state = {
+            relatedServices: [],  // includes us, use this.relatedServices
```
Comment (danni) [inline src/components/CrisisLineItem.js]: Don't need this.

File: src/components/CrisisLineItem.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        };
+    }
+
+    // flow:disable not supported yet
+    static propTypes = {
+        object: React.PropTypes.object.isRequired,
+        nRelatedServices: React.PropTypes.number,
```
Comment (danni) [inline src/components/CrisisLineItem.js]: Or this.

File: src/components/CrisisLineItem.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    };
+
+    // flow:disable not supported yet
+    static sampleProps = {object: fixtures.ixa};
```
Comment (danni) [inline src/components/CrisisLineItem.js]: Make your sample props be a crisis service.

File: src/components/CrisisLineItem.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+    // flow:disable
+    static defaultProps =  {
+        nRelatedServices: 4,
```
Comment (danni) [inline src/components/CrisisLineItem.js]: Don't need this.

File: src/components/CrisisLineItem.js [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    };
+
+    /**
+     * relatedServices:
+     * A list of related services, limited to 4
+     */
+    /* flow:disable */
+    get relatedServices(): Array<Object> {
```
Comment (danni) [inline src/components/CrisisLineItem.js]: Don't need this.

File: src/components/CrisisLineItem.js [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+        return this.state.relatedServices
+            .slice(0, this.props.nRelatedServices);
+    }
+
+    /**
+     * nMoreRelatedServices:
+     * The number of related services minus the 4 relatedServices.
+     */
+    /* flow:disable */
+    get nMoreRelatedServices(): number {
```
Comment (danni) [inline src/components/CrisisLineItem.js]: Don't need this.

File: src/components/CrisisLineItem.js [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+        return Math.max(0,
+                        this.state.relatedServices.length -
+                            this.props.nRelatedServices);
+    }
+
+    componentDidMount(): void {
+        /* request our sibling services */
```
Comment (danni) [inline src/components/CrisisLineItem.js]: Don't need this.

File: src/components/CrisisLineItem.js [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+
+                                <icons.Phone className="phoneIcon" />
+                                <div className="crisisItem">
+                                 {crisisType} {phone.number}</div>
+
+                        </mui>
+                    );
+            }
+        }
+
```
Comment (danni) [inline src/components/CrisisLineItem.js:50]: Needs to return something here. Currently returns void.

File: src/components/CrisisLineItem.scss [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+        border-radius: 3px;
+        width: 70%;
+        margin-left: 5%;
+        margin-bottom: 10px;
+        margin-right: 5%;
+        height: 32px;
+        padding-bottom: 20px;
+        font-size: 24px !important;
+        padding-left: 15%;
+        padding-top: 17px;
```
Comment (danni) [inline src/components/CrisisLineItem.scss]: Can we keep these sorted/grouped?

File: src/components/CrisisLineItem.scss [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+
+    }
+
+    .phoneIcon{
+            position: absolute;
+            width: 45px !important;
+            height: 30px !important;
+            padding: 0;
+            padding-top: 20px;
+            padding-right: 0%;
```
Comment (danni) [inline src/components/CrisisLineItem.scss]: Just 0.

File: src/components/CrisisLineItem.scss [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+            margin-left: 9%;
```
Comment (danni) [inline src/components/CrisisLineItem.scss]: Don't use a %age here.

File: src/components/CrisisLineItem.scss [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+        border-color: $brand-pane-alt-light;
+        border-radius: 3px;
+        width: 70%;
+        margin-left: 5%;
+        margin-bottom: 10px;
+        margin-right: 5%;
+        height: 32px;
+        padding-bottom: 20px;
+        font-size: 24px !important;
+        padding-left: 15%;
```
Comment (danni) [inline src/components/CrisisLineItem.scss]: Don't use a %age, use a fixed number of pixels.

File: src/components/CrisisLineItem.scss [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+            padding: 0;
+            padding-top: 20px;
+            padding-right: 0%;
+            margin-left: 9%;
+            color: $brand-pane-alt-light;
+
+        }
+    .crisisName{
+        color: $brand-text-mid;
+        margin-left:5% ;
```
Comment (danni) [inline src/components/CrisisLineItem.scss]: Also here.

File: src/pages/ResultsListPage.js [hunk 14]
```diff
@@ -135,7 +137,7 @@ class ResultsListPage extends React.Component {
 
     }
 
-    render(): React.Element {
+render(): React.Element {
```
Comment (danni) [inline src/pages/ResultsListPage.js]: Indenting?

File: src/pages/ResultsListPage.js [hunk 15]
```diff
@@ -198,10 +201,17 @@ class ResultsListPage extends React.Component {
                             <div key={index}>
                                 {React.addons.cloneWithProps(object.node)}
                             </div>
+                        : object.crisis ?
+
+                             <CrisisLineItem object={object} key={index} />
                         :
-                            <ResultListItem object={object} key={index} />
```
Comment (danni) [inline src/pages/ResultsListPage.js]: Indenting?

File: test/features/10-functional/10-basic/40-crisisLine.feature [hunk 16]
```diff
@@ -0,0 +1,22 @@
+Feature: Crisis Line
```
Comment (danni) [inline test/features/10-functional/10-basic/40-crisisLine.feature]: 40-crisis-line.feature please :)

File: test/features/10-functional/10-basic/40-crisisLine.feature [hunk 17]
```diff
@@ -0,0 +1,22 @@
...
+    # When I select any category
+    # I want to see results including the crisis phone numbers
+    # So that I can use the number to get help
+
+Scenario: Get helpline phone number for the food category
+        Given my location is "Melbourne VIC"
+        When I visit /
+        And I click on "Food"
+        Then I should be at /category/food
+        And I should see phone number 0311111111
```
Comment (danni) [inline test/features/10-functional/10-basic/40-crisisLine.feature]: `I should see hotline 03111111 in position 1`

**Comment by danni** [comment]
Please fix the spelling of your commit message also.

**Comment by pfctdayelise** [comment]
So this has the following fixes:
- Up to date with master (more or less)
- Makes phone number clickable
- Improves styling (look of styling if not coding of styling...)
- Addresses most PR comments

Still todo:
- Grey header/footer for crisis line sections
- Grey phone number on hover
- "If you need urgent help, please call one of these numbers" text
- Test that phone number is clickable/callable

File: src/components/CrisisLineItem.js [hunk 18]
```diff
@@ -0,0 +1,50 @@
...
+import fixtures from "../../fixtures/services";
+import mui from "material-ui";
+import reactMixin from "react-mixin";
+import _ from "underscore";
+import icons from "../icons";
+import iss from '../iss';
+
+var palette = colors.getPalette();
+
+/*::`*/@reactMixin.decorate(Router.Navigation)/*::`;*/
```
Comment (danni) [inline src/components/CrisisLineItem.js]: Not needed.

File: src/components/CrisisLineItem.js [hunk 19]
```diff
@@ -0,0 +1,50 @@
...
+import React from "react";
+import Router from "react-router";
+import Phone from "./Phone";
+import colors from "../constants/theme";
+import fixtures from "../../fixtures/services";
+import mui from "material-ui";
+import reactMixin from "react-mixin";
+import _ from "underscore";
+import icons from "../icons";
+import iss from '../iss';
```
Comment (danni) [inline src/components/CrisisLineItem.js:12]: Sort node_modules from our source. And then sort alphabetically.

File: src/components/CrisisLineItem.js [hunk 20]
```diff
@@ -0,0 +1,50 @@
...
+            object,
+        } = this.props;
+
+        for (var kind of ['freecall', 'phone', 'mobile']) {
+            var phone = _.findWhere(object.phones, {kind: kind});
+
+            if (phone) {
+                return (
+                    <div className="CrisisLineItem">
+                        <div className="crisisName"> {object.name}
```
Comment (danni) [inline src/components/CrisisLineItem.js]: Use a header tag so it comes up nicely in a screen reader.

File: src/components/CrisisLineItem.js [hunk 21]
```diff
@@ -0,0 +1,50 @@
...
+                        </div>
+                        <Phone {...phone} />
```
Comment (danni) [inline src/components/CrisisLineItem.js:45]: Awesome code reuse :)

File: src/components/CrisisLineItem.scss [hunk 22]
```diff
@@ -0,0 +1,43 @@
...
+
+        display: block;
+        position: relative;
+	width: auto;
+
+        border: 1px solid;
+        border-color: $brand-pane-alt-dark;
+        border-radius: 3px;
+
+	padding: 6px;
```
Comment (danni) [inline src/components/CrisisLineItem.scss]: Weird indenting in this file.

File: src/pages/ResultsListPage.js [hunk 23]
```diff
@@ -8,6 +8,8 @@ import mui from "material-ui";
 
 import BaseResultsPage from "./BaseResultsPage";
 import ResultListItem from '../components/ResultListItem';
+import CrisisLineItem from '../components/CrisisLineItem';
+import HeaderBar from '../components/HeaderBar';
```
Comment (danni) [inline src/pages/ResultsListPage.js]: Already available as `components.` also I can't see where you're using them. Merge artifact?

File: test/features/10-functional/10-basic/40-crisis-line.feature [hunk 24]
```diff
@@ -0,0 +1,24 @@
...
+    Background:
+        Given my location is "Melbourne VIC"
+        And my gender is female
+        And I am not part of any relevant demographics
+
+    Scenario: Get helpline phone number for the food category
+        When I visit /
+        And I click on "Food"
+        Then I should be at /category/food
+        And I should see phone number Mobile 0311111111
```
Comment (danni) [inline test/features/10-functional/10-basic/40-crisis-line.feature]: `I should see hotline in position 1 with mobile 0311111111`

File: test/features/10-functional/10-basic/40-crisis-line.feature [hunk 25]
```diff
@@ -0,0 +1,24 @@
...
+
+    Scenario: Search for helpline phone numbers for food
+        When I visit /
+        And I search for "food"
+        And I click on the search icon
+        Then I should see phone number Mobile 0311111111
```
Comment (danni) [inline test/features/10-functional/10-basic/40-crisis-line.feature]: And here.

**Comment by danni** [comment]
Be good to use your domestic violence service in a test.

**Comment by pfctdayelise** [comment]
So I added my fixture to `fixtures/server.js`, not `test/support/mock_iss/server.js`...should I add it in both places? Should the fixtures be automatically available in the test fixtures?
Also I still haven't tested that the phone number is clickable #TODO

**Comment by DanielHeath** [comment]
To make the fixture available to tests, add something like the following to `mock_iss/server.js`: 

```
app.get('/api/v3/service/537512/', (req, res) => {
    res.json(services.domesticviolence);
});
```

There's a bunch of others like it near the end of the file.

Then in your feature you can do
`When I visit /service/537512`

**Comment by pfctdayelise** [comment]
...But it wouldn't show up in the search results? This feature is only about how the results list looks.

**Comment by DanielHeath** [comment]
++

**Comment by pfctdayelise** [comment]
😎😎😎

**Pull request merged**